rpc/core/tx.go

```diff
@@ -19,36 +19,44 @@ import (
 // More: https://docs.tendermint.com/master/rpc/#/Info/tx
 func (env *Environment) Tx(ctx *rpctypes.Context, hash []byte, prove bool) (*ctypes.ResultTx, error) {
 	// if index is disabled, return error
-	if _, ok := env.TxIndexer.(*null.TxIndex); ok {
-		return nil, fmt.Errorf("transaction indexing is disabled")
-	}
 
-	r, err := env.TxIndexer.Get(hash)
-	if err != nil {
-		return nil, err
+	if !indexer.KVSinkEnabled(env.EventSinks) {
+		return nil, errors.New("transaction querying is disabled due to no kvEventSink")
 	}
 
-	if r == nil {
-		return nil, fmt.Errorf("tx (%X) not found", hash)
-	}
+	for _, sink := range env.EventSinks {
+		if sink.Type() == indexer.KV {
+			r, err := sink.GetTxByHash(hash)
+			if err != nil {
+				return nil, err
+			}
+
+			if r == nil {
+				return nil, fmt.Errorf("tx (%X) not found", hash)
```

Comment:
seems like it might make sense to just force/expect `GetTxByHash` to error in the not-found case.